chore: rewrite documents and split out sections (#117)
IBM/cloudant-kafka-connector: README.md

[![Maven Central](https://img.shields.io/maven-central/v/com.cloudant/kafka-connect-cloudant.svg)](http://search.maven.org/#search|ga|1|g:"com.cloudant"%20AND%20a:"kafka-connect-cloudant")

This project includes [Apache Kafka](https://kafka.apache.org/) [Connect](https://kafka.apache.org/documentation.html#connect) source and sink connectors for IBM Cloudant.

These connectors can stream events:
- **from** Cloudant (source connector) to Kafka topic(s)

> - **to** Cloudant (sink connector) from Kafka topic(s)

Reviewer: I think maybe worth adding a note that it is also compatible with Apache CouchDB.
```suggestion
- **to** Cloudant (sink connector) from Kafka topic(s)

_Note:_ the connectors are also compatible with Apache CouchDB.
```

Experimental

## Usage

**Note**: The below instructions assume an installation of Kafka at `$KAFKA_HOME`.

### Quick Start

1. Download the jar from the [releases page](https://github.com/IBM/cloudant-kafka-connector/releases). The jar file contains the plugin and the non-Kafka dependencies needed to run.
2. Copy the jar to the `libs` directory _or_ the
[configured `plugin.path`](https://kafka.apache.org/documentation.html#connectconfigs_plugin.path) of your Kafka installation.
3. Edit the [source](docs/connect-cloudant-source-example.properties) or [sink](docs/connect-cloudant-sink-example.properties) example properties files and save this to the `config` directory of your Kafka installation.
4. Start Kafka.
5. Start the connector (see below).

Connector execution in Kafka is available through scripts in the Kafka install path:

`$KAFKA_HOME/bin/connect-standalone.sh` or `$KAFKA_HOME/bin/connect-distributed.sh`

Use the appropriate configuration files for standalone or distributed execution with Cloudant as source, as sink, or both.

For example:

> - standalone execution with Cloudant as source:

Reviewer: ```suggestion
- standalone execution with Cloudant changes feed as source:
```

  ```

- standalone execution with Cloudant as sink:

  ```
  $KAFKA_HOME/bin/connect-standalone.sh \
  $KAFKA_HOME/config/connect-standalone.properties \
  $KAFKA_HOME/config/connect-cloudant-sink.properties
  ```

- standalone execution with multiple configurations, one using Cloudant as source and one using Cloudant as sink:

  ```
  $KAFKA_HOME/bin/connect-standalone.sh \
  $KAFKA_HOME/config/connect-standalone.properties \
  $KAFKA_HOME/config/connect-cloudant-source.properties \
  $KAFKA_HOME/config/connect-cloudant-sink.properties
  ```

Any number of connector configurations can be passed to the executing script.

## Configuration

As outlined above, the Cloudant Kafka connector can be configured in standalone or distributed mode according to 
the [Kafka Connector documentation](https://kafka.apache.org/documentation.html#connect_configuring).

The `connect-standalone` or `connect-distributed` configuration files contain default values which are necessary for all connectors, such as:

1. `bootstrap.servers`
2. If using a standalone worker `offset.storage.file.filename`.
3. `offset.flush.interval.ms`

### Connector configuration

> The `cloudant-source-example` and `cloudant-sink-example` properties files contain the minimum required to get started.

Reviewer: Suggest we make the source "type" more clear and link to the properties files
```suggestion
The [`cloudant-changes-source-example`](docs/connect-cloudant-changes-source-example.properties) and [`cloudant-sink-example`](docs/connect-cloudant-sink-example.properties) properties files contain the minimum required to get started.
```

> For a full reference explaining all the connector options, see [here (source)](docs/configuration-reference-source.md) and

Reviewer: ```suggestion
For a full reference explaining all the connector options, see [here (source)](docs/configuration-reference-changes-source.md) and
```

[here (sink)](docs/configuration-reference-sink.md).

#### Authentication

In order to read from or write to Cloudant, some authentication properties need to be configured. These properties are common to both the source and sink connector, and are detailed in the configuration reference, linked above.

> A number of different authentication methods are supported. IAM authentication is the default and recommended method; see [locating your service credentials](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-locating-your-service-credentials) for details on how to find your IAM API key.

Reviewer: I think any of `iam` `container` or `vpc` could be recommended as "IAM-based" so maybe a little re-wording here.

```suggestion
A number of different authentication methods are supported. IBM Cloud IAM-based authentication methods are recommended and the default is to use an IAM API key. See [locating your service credentials](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-locating-your-service-credentials) for details on how to find your IAM API key.
```
